src/array.rs

```diff
@@ -1480,6 +1480,21 @@ unsafe fn clone_elements<T: Element>(elems: &[T], data_ptr: &mut *mut T) {
     }
 }
 
+/// Implementation of functionality for [`PyArray<T, D>`].
+#[doc(alias = "PyArray")]
+// TODO: seal this trait
```

Comment:
Note that we already have `crate::sealed::Sealed` for that.

Reply:
I think we might want to be a bit careful here, because that `Sealed` already has a generic implementation for `IntoDimension`, which is not sealed. So using that would allow implementing `PyUntypedArrayMethods` for downstream types that also implement `IntoDimension`.

Reply:
I see. In that case we probably just want to split `Sealed` into at least four modules/traits for `convert`, `array_like`, `array` and `untyped_array`. (Better to have dedicated traits everywhere than to figure which could be shared and which should not be.)

Reply:
Sure, I'll add the one for `untyped_array` here.